enhancements/machine-api/aws-efa-networking.md

```diff
@@ -0,0 +1,253 @@
+between the hosts within the security group. This is not configured by default on an IPI cluster.
+Users must manually make this change via the AWS console/cli.
+This prerequisite should be documented as a manual step to enable the feature.
+
+Additionally, as we have future plans to manage security groups for IPI clusters, these manual changes to the security
+groups may be difficult to adopt into the future management system. However, there is nothing to stop users of IPI
+clusters making changes such as this already, so this kind of change must be considered for any adoption plans already.
+
+Additionally, IPI users are free to add additional security groups to their clusters post install and specify these
+additional groups within their Machine specs. We have no control over this today and again, this is something that must
+be planned for in the future security group management process.
+
+#### Using EFA requires additional software
+
+The EFA network interface types require special drivers to enable them to be used.
+Fortunately, the RHEL kernel already includes the relevant kernel modules to allow the adapter to be used transparently
+as a standard network adapter within user space. This means that no additional configuration is required within RHCOS
+to allow EFA instances to be used as regular OpenShift nodes, existing networking is unaffected.
+
+To take advantage of the EFA performance benefits, additional libfabric software must be deployed.
+This software is currently [distributed by AWS](https://github.com/aws-samples/aws-efa-eks/blob/main/manifest/efa-k8s-device-plugin.yml)
+and it will be up to the end user to deploy this software within their clusters.
+Red Hat will not initially support the libfabric driver.
+
+#### Only a subset of instance types are supported with EFA network interfaces
+
+Only a small subset of AWS instance types (eg. m5dn.24xlarge, m5dn.metal) are supported for use with EFA interfaces.
+If a user were to specify an EFA interface type with a non supported instance (eg. m5.xlarge), then the instance will
+fail to create. In this scenario, AWS rejects the RunInstance call from Machine API which causes Machine API to mark the
+Machine as `Failed`. An error message will be added to the Machine status to signal to the user that their configuration
+was invalid.
+
+> error launching instance: EFA interfaces are not supported on m5.2xlarge
+
+## Design Details
+
+### Open Questions
+
+- What support will we provide in the future?
+  - Will we be able to provide full end to end support for the required libfabric drivers?
+- How can the feature be used by end users?
+  - How do they signal to OpenShift that a particular workload should leverage the EFA feature?
+
+### Test Plan
+
+In the first phase, we are planning to just provide users the ability to create Machines with the EFA interface type.
+We will run standard regression testing on the feature but we do not expect any special additional testing to be
```

Comment:
I could probably buy "no automated testing", though I don't think that's a great story.  But we should make sure the feature actually succeeds at its goal of enabling this EFA thing in a way that a workload can actually  use.